WcaOnRails/app/models/incident.rb

```diff
@@ -38,4 +38,37 @@ def digest_sent_at_consistent
       errors.add(:digest_sent_at, "can't be set if incident is not resolved.")
     end
   end
+
+  DEFAULT_PUBLIC_SERIALIZE_OPTIONS = {
+    only: [:id, :title, :public_summary, :created_at, :updated_at, :resolved_at],
+  }.freeze
+
+  DEFAULT_DELEGATE_MATTERS_SERIALIZE_OPTIONS = {
+    only: DEFAULT_PUBLIC_SERIALIZE_OPTIONS[:only] +
+      [:private_description, :digest_worthy, :digest_sent_at],
+  }.freeze
+
+  def serializable_hash(options = nil)
+    if (options && options[:can_view_delegate_matters])
+      options = DEFAULT_DELEGATE_MATTERS_SERIALIZE_OPTIONS.merge(options || {})
+    else
+      options = DEFAULT_PUBLIC_SERIALIZE_OPTIONS.merge(options || {})
+    end
+
+    json = super(options)
+
+    json[:tags] = tags_array.map { |tag|
+      { name: tag }.merge(Regulation.find_or_nil(tag) || {})
+    }
+
+    json[:competitions] = incident_competitions.map { |incident_competition|
+      {
+        id: incident_competition.competition.id,
+        name: incident_competition.competition.name,
+        comments: incident_competition.comments,
+      }
+    }
```

Comment:
This may (or may not, honestly unsure) be possible to do more elegantly in a way similar to what Philippe is proposing over in https://github.com/thewca/worldcubeassociation.org/pull/7199

Quote from his description as a sample: (note the `include` part towards the end)
> `competition.serializable_hash(only: ["name"], methods: ["country_iso2"], include: { organizers: { only: ["id", "name"] })`

Reply:
I tried that, but then you only have access - as far as I could figure out (which isn't very far because everything is implicit) - to `incident_competition.competition` attributes, but not `incident_competition.comments`.

Similarly for the tags, I don't see how to 'find the regulation (or nil)' within the `include`.

Reply:
Okay! You can decide yourself whether you want to read up some more and play round with the intrinsics of `serializable_hash`. Like above, I'm really only pointing out ideas for you to consider :)

Reply:
I'm probably not willing to look into this any further than I already tried, at this point. But thanks for the suggestion.

Reply:
Sorry I did not see this earlier, this is probably what you want (I've taken an arbitrary incident just for the sake of the example):
```ruby
irb(main):014:0> Incident.find(55).incident_competitions.as_json(only: ["comments"], include: { competition: { only: ["id", "name"], methods:[], include: [] }})
[{"comments"=>nil, "competition"=>{"id"=>"AltonaAlgorithmsAttempt22021", "name"=>"Altona Algorithms Attempt 2 2021", "class"=>"competition"}},
 {"comments"=>nil, "competition"=>{"id"=>"GetafeContinua2020", "name"=>"Getafe Continúa 2020", "class"=>"competition"}},
 {"comments"=>nil, "competition"=>{"id"=>"OsijekOpen2020", "name"=>"Osijek Open 2020", "class"=>"competition"}}]
```

("comments" is nil here because the output comes from the dev dump)

For the tag thing: indeed it's quite hard to get it from there, one approach would be to "nest" the regulation within, and have something like this: `{ name: "somename", regulation: { /* reg attributes */ } }`; in which case it should be as simple as implementing a `regulation` method on `IncidentTag` which does the `Regulation.find_or_nil` you did.
Anyway this is not a big deal.